dataprofiler/profilers/base_column_profilers.py

```diff
@@ -320,4 +320,4 @@ def _add_helper(  # type: ignore[override]
         :type other2: BaseColumnPrimitiveTypeProfiler
         """
         BaseColumnProfiler._add_helper(self, other1, other2)
-        self.match_count = other1.match_count + other2.match_count
+        self.match_count = int(other1.match_count + other2.match_count)
```

Comment:
Cast added to adhere to type specification in numeric stats mixin attribute initialization (avoids setting to np.int64)